sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -139,6 +139,23 @@ case class CollectSet(
 
   def this(child: Expression) = this(child, 0, 0)
 
+  override def eval(buffer: mutable.HashSet[Any]): Any = {
```

Comment:
I don't think this is how we should fix this. `eval(..)` is called super late in the game, and all that time we are keeping duplicates in memory and we are shuffling them. How about converting binary to an `UnsafeArrayData` object in `update()`? `UnsafeArrayData` has a decent `equals` and `hashCode` implementation so that should work. You will need to convert them back in eval.

Reply:
yes, I thought about it, and I was trying to solve it in previous fases but I found problems with the types. 
I think the key is `UnsafeArrayData` object. I will try at this way. Thank you very much for the help!